File: src/test/java/de/moritzf/quota/idea/QuotaDisplayModeTest.java
package de.moritzf.quota.idea;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class QuotaDisplayModeTest {
    @Test
    void fromStorageValueFallsBackToIconOnly() {
        assertEquals(QuotaDisplayMode.ICON_ONLY, QuotaDisplayMode.fromStorageValue(null));
        assertEquals(QuotaDisplayMode.ICON_ONLY, QuotaDisplayMode.fromStorageValue(""));
        assertEquals(QuotaDisplayMode.ICON_ONLY, QuotaDisplayMode.fromStorageValue("unknown"));
    }

    @Test
    void fromStorageValueParsesAllModesCaseInsensitively() {
        assertEquals(QuotaDisplayMode.ICON_ONLY, QuotaDisplayMode.fromStorageValue("ICON_ONLY"));
        assertEquals(QuotaDisplayMode.PERCENTAGE_BAR, QuotaDisplayMode.fromStorageValue("percentage_bar"));
        assertEquals(QuotaDisplayMode.CAKE_DIAGRAM, QuotaDisplayMode.fromStorageValue("cake_diagram"));
    }
}

File: src/test/kotlin/de/moritzf/quota/idea/QuotaDisplayModeTest.kt
package de.moritzf.quota.idea

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class QuotaDisplayModeTest {
    @Test
    fun fromStorageValueFallsBackToIconOnly() {
        assertEquals(QuotaDisplayMode.ICON_ONLY, QuotaDisplayMode.fromStorageValue(null))
        assertEquals(QuotaDisplayMode.ICON_ONLY, QuotaDisplayMode.fromStorageValue(""))
        assertEquals(QuotaDisplayMode.ICON_ONLY, QuotaDisplayMode.fromStorageValue("unknown"))
    }

    @Test
    fun fromStorageValueParsesAllModesCaseInsensitively() {
        assertEquals(QuotaDisplayMode.ICON_ONLY, QuotaDisplayMode.fromStorageValue("ICON_ONLY"))
        assertEquals(QuotaDisplayMode.PERCENTAGE_BAR, QuotaDisplayMode.fromStorageValue("percentage_bar"))
        assertEquals(QuotaDisplayMode.CAKE_DIAGRAM, QuotaDisplayMode.fromStorageValue("cake_diagram"))
    }

    @Test
    fun mainToolbarOnlySupportsIconAndCakeModes() {
        val supported = QuotaDisplayMode.supportedFor(QuotaIndicatorLocation.MAIN_TOOLBAR)
        assertTrue(QuotaDisplayMode.ICON_ONLY in supported)
        assertTrue(QuotaDisplayMode.CAKE_DIAGRAM in supported)
        assertFalse(QuotaDisplayMode.PERCENTAGE_BAR in supported)
    }

    @Test
    fun sanitizeForMainToolbarFallsBackFromPercentageBar() {
        assertEquals(
            QuotaDisplayMode.ICON_ONLY,
            QuotaDisplayMode.sanitizeFor(QuotaIndicatorLocation.MAIN_TOOLBAR, QuotaDisplayMode.PERCENTAGE_BAR),
        )
    }
}
